dispatches/models/nuclear_case/chgtank.py

```diff
@@ -0,0 +1,590 @@
+            within=NonNegativeReals,
+            initialize=1.0,
+            doc="Length of storage tank in m",
+            units=pyunits.m)
+
+        # Tank volume calculation
+        @self.Constraint(self.flowsheet().config.time)
+        def volume_cons(b, t):
+            return (b.control_volume.volume[t] ==
+                    const.pi * b.tank_length[t] *
+                    ((b.tank_diameter[t]/2)**2))
+
+        # define Vars for the model
+        self.dt = Var(
+            self.flowsheet().config.time,
+            domain=NonNegativeReals,
+            initialize=100,
+            doc="time step for holdup calculation",
+            units=time_units)
+
+        self.heat_duty = Var(
+            self.flowsheet().config.time,
+            domain=Reals,
+            initialize=0.0,
+            doc="Heat transfered through walls, is 0 for adiabatic",
+            units=power_units)
+
+        self.material_accumulation = Var(
+            self.flowsheet().config.time,
+            pc_set,
+            within=Reals,
+            initialize=1.0,
+            doc="Accumulation of material in tank",
+            units=flow_units)
+
+        self.energy_accumulation = Var(
+            self.flowsheet().config.time,
+            phase_list,
+            within=Reals,
+            initialize=1.0,
+            doc="Energy accumulation",
+            units=power_units)
+
+        self.material_holdup = Var(
+            self.flowsheet().config.time,
+            pc_set,
+            within=Reals,
+            initialize=1.0,
+            doc="Material holdup in tank",
+            units=flow_units*time_units)
+
+        self.energy_holdup = Var(
+            self.flowsheet().config.time,
+            phase_list,
+            within=Reals,
+            initialize=1.0,
+            doc="Energy holdup in tank",
+            units=energy_units)
+
+        self.initial_material_holdup = Var(
+            self.flowsheet().config.time,
+            pc_set,
+            within=Reals,
+            initialize=1.0,
+            doc="Initial material holdup in tank",
+            units=flow_units*time_units)
+
+        self.initial_energy_holdup = Var(
+            self.flowsheet().config.time,
+            phase_list,
+            within=Reals,
+            initialize=1.0,
+            doc="Initial energy holdup in tank",
+            units=energy_units)
+
+        # Adiabatic operations are assumed
+        # Fixing the heat_duty to 0 here to avoid any misakes at use
+        # To Do: remove this once the isothermal constraints are added
+        self.heat_duty.fix(0)
+
+        # Computing initial material and energy holdup in the tank 
+        # using initial state Pressure and Temperature of the tank
+        # Assuming Ideal gas law 
+        @self.Constraint(self.flowsheet().config.time,
+                         phase_list,
+                         doc="Initial material holdup")
+        def initial_material_holdup_rule(b, t, p):
```

Comment:
1. First of all, shouldn't the initial holdup be based on the holdup at the previous step.
2. At the initial time point, you should be using the actual property package rather than assuming ideal.